src/liquidator.ts

```diff
@@ -343,17 +344,26 @@ async function newAccountOnLiquidatableFeed(params) {
 
 // never returns
 async function liquidatableFromLiquidatableFeed() {
+  let candidates = new AsyncBlockingQueue<string>();
+  let candidatesSet = new Set<string>();
   const ws = new RpcWebSocketClient(liquidatableFeedWebsocketAddress, {
     max_reconnects: Infinity,
   });
   ws.on('open', (x) => console.log("opened liquidatable feed"));
   ws.on('error', (status) => console.log("error on liquidatable feed", status));
   ws.on('close', (err) => console.log("closed liquidatable feed", err));
-  ws.on('liquidatable', newAccountOnLiquidatableFeed);
+  ws.on('candidate', (params) => {
+      const account = params.account;
+      if (!candidatesSet.has(account)) {
+        candidatesSet.add(account);
+        candidates.enqueue(account);
+      }
+  });
 
-  // fully reactive now
   while (true) {
-    await sleep(5000);
+    const account = await candidates.dequeue();
```

Comment:
what if the queue is empty? then it returns an empty promise? but then that is not a mango account, or?

Reply:
If the queue is empty this blocks until a new element is added, then returns it. (it returns a promise that will only be resolved by the next `enqueue()`)